src/e2e-test/java/io/cdap/plugin/gcscopy/stepsdesign/GCSCopy.java

```diff
@@ -0,0 +1,158 @@
+                                          + "/" + PluginPropertyUtils.pluginProp(path));
+  }
+
+  @Then("Validate GCSCopy successfully copied object {string} to destination bucket in location {string}")
+  public void validateGCSCopySuccessfullyCopiedObjectToDestinationBucketInLocation(String path, String location) {
+    String targetGCSBucket = TestSetupHooks.gcsTargetBucketName;
+    try {
+      if (StorageClient.getBucketMetadata(targetGCSBucket).getLocation().equalsIgnoreCase(PluginPropertyUtils.pluginProp
+        (location))) {
+        validateGCSCopySuccessfullyCopiedObjectToDestinationBucket(path);
+        return;
+      }
+      Assert.fail("Target gcs bucket " + targetGCSBucket + " is not created in location " + location);
+    } catch (StorageException | IOException e) {
+      if (e.getMessage().contains("The specified bucket does not exist")) {
+        Assert.fail("Target gcs bucket " + targetGCSBucket + " not created - " + e.getMessage());
+      } else {
+        Assert.fail(e.getMessage());
+      }
+    }
+  }
+
+  @Then("Enter runtime argument value {string} for GCSCopy property sourcePath key {string}")
+  public void enterRuntimeArgumentValueForGCSCopyPropertySourcePathKey(String value, String runtimeArgumentKey) {
+    ElementHelper.sendKeys(CdfStudioLocators.runtimeArgsValue(runtimeArgumentKey),
+                           "gs://" + TestSetupHooks.gcsSourceBucketName + "/" + PluginPropertyUtils.pluginProp(value));
+  }
+
+  @Then("Enter runtime argument value for GCSCopy property destination path key {string}")
+  public void enterRuntimeArgumentValueForGCSCopyPropertyDestinationPathKey(String runtimeArgumentKey) {
+    ElementHelper.sendKeys(CdfStudioLocators.runtimeArgsValue(runtimeArgumentKey),
+                           "gs://" + TestSetupHooks.gcsTargetBucketName);
+  }
+
+  @Then("Validate GCSCopy did not copy subdirectory {string} to destination bucket")
+  public void validateGCSCopyDidNotCopySubdirectoryToDestinationBucket(String subdirectory) throws IOException {
+    String sourceGCSBucket = TestSetupHooks.gcsSourceBucketName;
+    String gcsObject = PluginPropertyUtils.pluginProp(subdirectory);
+    for (Blob blob : StorageClient.listObjects(sourceGCSBucket).iterateAll()) {
+      if (blob.getName().contains(gcsObject)) {
+
+        BeforeActions.scenario.write("Subdirectory is not copied from source bucket" + sourceGCSBucket);
+        return;
+      }
+    }
+      Assert.fail("Subdirectory is copied from source GCS Bucket " + sourceGCSBucket);
+    }
+
+  @Then("Validate GCSCopy failed to copy object {string} to destination bucket")
+  public void validateGCSCopyFailedToCopyObjectToDestinationBucket(String path) throws IOException {
+    String sourceGCSBucket = TestSetupHooks.gcsSourceBucketName;
+    String gcsObject = PluginPropertyUtils.pluginProp(path);
+    for (Blob blob : StorageClient.listObjects(sourceGCSBucket).iterateAll()) {
+      if (blob.getName().contains(gcsObject)) {
```

Comment:
why is this a contains check and not an equals check ?

Reply:
Changed to equals check.